### ISSUE TITLE
freeorion fails to build with boost 1.66.0

### ISSUE THREAD

**Issue by kyrios123** [freeorion/freeorion#1967]
### Environment

* **FreeOrion Version**: 0.4.7.1
* **Operating System**: x86_64 GNU/Linux (Solus 4.14.15-48.current)
* **Fetched as**
  * Compiled from source

### Description
```
# make -j4
Scanning dependencies of target freeorionversion
WARNING: git not installed or not setup correctly
WARNING: Can't determine git commit!
Writing file: util/Version.cpp
Building v0.4.7.1 build ???
[  0%] Built target freeorionversion
Scanning dependencies of target GiGi
[  0%] Building CXX object GG/CMakeFiles/GiGi.dir/src/dialogs/ColorDlg.cpp.o
[  0%] Building CXX object GG/CMakeFiles/GiGi.dir/src/dialogs/FileDlg.cpp.o
Scanning dependencies of target freeorionparseobj
[  0%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/AlignmentsParser.cpp.o
Scanning dependencies of target freeorioncommon
[  0%] Building CXX object CMakeFiles/freeorioncommon.dir/combat/CombatEvent.cpp.o
[  1%] Building CXX object GG/CMakeFiles/GiGi.dir/src/dialogs/ThreeButtonDlg.cpp.o
[  2%] Building CXX object CMakeFiles/freeorioncommon.dir/combat/CombatEvents.cpp.o
[  3%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/BuildingsParser.cpp.o
[  3%] Building CXX object GG/CMakeFiles/GiGi.dir/src/RichText/BlockControl.cpp.o
[  3%] Building CXX object GG/CMakeFiles/GiGi.dir/src/RichText/ImageBlock.cpp.o
[  3%] Building CXX object CMakeFiles/freeorioncommon.dir/combat/CombatLogManager.cpp.o
[  3%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/CommonParamsParser.cpp.o
[  3%] Building CXX object CMakeFiles/freeorioncommon.dir/Empire/Diplomacy.cpp.o
[  4%] Building CXX object CMakeFiles/freeorioncommon.dir/Empire/Empire.cpp.o
[  5%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser1.cpp.o
/home/build/YPKG/root/freeorion/build/src-tarball/Empire/Empire.cpp:1395:29: warning: ‘const {anonymous}::AlignmentManager& {anonymous}::GetAlignmentManager()’ defined but not used [-Wunused-function]
     const AlignmentManager& GetAlignmentManager() {
                             ^~~~~~~~~~~~~~~~~~~
[  5%] Building CXX object CMakeFiles/freeorioncommon.dir/Empire/EmpireManager.cpp.o
[  5%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser2.cpp.o
[  6%] Building CXX object CMakeFiles/freeorioncommon.dir/Empire/ResourcePool.cpp.o
[  6%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser3.cpp.o
[  6%] Building CXX object CMakeFiles/freeorioncommon.dir/Empire/Supply.cpp.o
[  6%] Building CXX object CMakeFiles/freeorioncommon.dir/network/Message.cpp.o
[  7%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser4.cpp.o
[  8%] Building CXX object CMakeFiles/freeorioncommon.dir/network/MessageQueue.cpp.o
[  8%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser5.cpp.o
[  8%] Building CXX object CMakeFiles/freeorioncommon.dir/network/Networking.cpp.o
[  9%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Building.cpp.o
[ 10%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser6.cpp.o
[ 10%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Condition.cpp.o
[ 10%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser7.cpp.o
[ 10%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/EffectAccounting.cpp.o
[ 11%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Effect.cpp.o
[ 11%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ConditionParser.cpp.o
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp: In member function ‘virtual void Effect::Conditional::Execute(const ScriptingContext&, const TargetSet&) const’:
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3371:90: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         Effect::TargetSet& match_targets = *reinterpret_cast<Effect::TargetSet*>(&matches);
                                                                                          ^
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3378:98: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         Effect::TargetSet& non_match_targets = *reinterpret_cast<Effect::TargetSet*>(&non_matches);
                                                                                                  ^
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp: In member function ‘void Effect::Conditional::Execute(const ScriptingContext&, const TargetSet&, Effect::AccountingMap*, bool, bool, bool, bool) const’:
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3432:90: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         Effect::TargetSet& match_targets = *reinterpret_cast<Effect::TargetSet*>(&matches);
                                                                                          ^
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3446:98: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         Effect::TargetSet& non_match_targets = *reinterpret_cast<Effect::TargetSet*>(&non_matches);
                                                                                                  ^
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp: In member function ‘virtual void Effect::Conditional::Execute(const TargetsCauses&, Effect::AccountingMap*, bool, bool, bool, bool) const’:
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3480:145: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
         const Condition::ObjectSet& potential_target_objects = *reinterpret_cast<const Condition::ObjectSet*>(&(targets_entry.second.target_set));
                                                                                                                                                 ^
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3488:94: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
             Effect::TargetSet& match_targets = *reinterpret_cast<Effect::TargetSet*>(&matches);
                                                                                              ^
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Effect.cpp:3502:102: warning: dereferencing type-punned pointer will break strict-aliasing rules [-Wstrict-aliasing]
             Effect::TargetSet& non_match_targets = *reinterpret_cast<Effect::TargetSet*>(&non_matches);
                                                                                                      ^
[ 11%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Encyclopedia.cpp.o
[ 12%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/DoubleComplexValueRefParser.cpp.o
[ 13%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Enums.cpp.o
[ 13%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Field.cpp.o
[ 13%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/DoubleValueRefParser.cpp.o
[ 13%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Fighter.cpp.o
[ 14%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EffectParser1.cpp.o
[ 15%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Fleet.cpp.o
[ 15%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Meter.cpp.o
[ 15%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EffectParser2.cpp.o
[ 16%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/ObjectMap.cpp.o
[ 16%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Planet.cpp.o
[ 16%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EffectParser3.cpp.o
[ 16%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/PopCenter.cpp.o
[ 17%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EffectParser4.cpp.o
[ 18%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Predicates.cpp.o
[ 18%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/ResourceCenter.cpp.o
[ 18%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EffectParser5.cpp.o
[ 19%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Ship.cpp.o
[ 19%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EffectParser.cpp.o
[ 19%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/ShipDesign.cpp.o
[ 20%] Building CXX object GG/CMakeFiles/GiGi.dir/src/RichText/RichText.cpp.o
[ 21%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EmpireStatsParser.cpp.o
[ 21%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Special.cpp.o
[ 21%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EncyclopediaParser.cpp.o
[ 22%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Species.cpp.o
[ 23%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/EnumParser.cpp.o
[ 23%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/System.cpp.o
/home/build/YPKG/root/freeorion/build/src-tarball/GG/src/dialogs/FileDlg.cpp: In member function ‘void GG::FileDlg::OpenDirectory()’:
/home/build/YPKG/root/freeorion/build/src-tarball/GG/src/dialogs/FileDlg.cpp:748:48: error: ‘boost::system::posix_error’ has not been declared
                 if (e.code() == boost::system::posix_error::io_error) {
                                                ^~~~~~~~~~~
make[2]: *** [GG/CMakeFiles/GiGi.dir/build.make:87: GG/CMakeFiles/GiGi.dir/src/dialogs/FileDlg.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
[ 23%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Tech.cpp.o
[ 24%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Universe.cpp.o
[ 24%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/Pathfinder.cpp.o
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Universe.cpp: In function ‘void {anonymous}::ShareVisbilitiesBetweenAllies(Universe::EmpireObjectVisibilityMap&, Universe::EmpireObjectSpecialsMap&)’:
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Universe.cpp:2285:46: warning: unused variable ‘allied_obj_specials_map’ [-Wunused-variable]
                 Universe::ObjectSpecialsMap& allied_obj_specials_map = input_eovs_copy[allied_empire_id];
                                              ^~~~~~~~~~~~~~~~~~~~~~~
/home/build/YPKG/root/freeorion/build/src-tarball/universe/Universe.cpp:2275:42: warning: unused variable ‘obj_specials_map’ [-Wunused-variable]
             Universe::ObjectSpecialsMap& obj_specials_map = empire_object_visible_specials[empire_id];
                                          ^~~~~~~~~~~~~~~~
[ 24%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/FieldsParser.cpp.o
[ 25%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/UniverseObject.cpp.o
[ 25%] Building CXX object CMakeFiles/freeorioncommon.dir/universe/ValueRef.cpp.o
[ 25%] Building CXX object CMakeFiles/freeorioncommon.dir/util/AppInterface.cpp.o
[ 26%] Building CXX object CMakeFiles/freeorioncommon.dir/util/Directories.cpp.o
[ 26%] Building CXX object CMakeFiles/freeorioncommon.dir/util/EnumText.cpp.o
[ 26%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/FleetPlansParser.cpp.o
[ 27%] Building CXX object CMakeFiles/freeorioncommon.dir/util/i18n.cpp.o
[ 27%] Building CXX object CMakeFiles/freeorioncommon.dir/util/Logger.cpp.o
[ 27%] Building CXX object CMakeFiles/freeorioncommon.dir/util/ModeratorAction.cpp.o
[ 28%] Building CXX object CMakeFiles/freeorioncommon.dir/util/MultiplayerCommon.cpp.o
[ 28%] Building CXX object CMakeFiles/freeorioncommon.dir/util/OptionsDB.cpp.o
[ 29%] Building CXX object CMakeFiles/freeorioncommon.dir/util/Order.cpp.o
[ 30%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/IntComplexValueRefParser.cpp.o
[ 30%] Building CXX object CMakeFiles/freeorioncommon.dir/util/OrderSet.cpp.o
[ 30%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/IntValueRefParser.cpp.o
[ 31%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ItemsParser.cpp.o
[ 31%] Building CXX object CMakeFiles/freeorioncommon.dir/util/Process.cpp.o
[ 32%] Building CXX object CMakeFiles/freeorioncommon.dir/util/Random.cpp.o
[ 32%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/KeymapParser.cpp.o
[ 32%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/Lexer.cpp.o
[ 32%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SaveGamePreviewUtils.cpp.o
/home/build/YPKG/root/freeorion/build/src-tarball/parse/Lexer.cpp: In constructor ‘parse::lexer::lexer()’:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/Lexer.cpp:30:1: note: variable tracking size limit exceeded with -fvar-tracking-assignments, retrying without
 lexer::lexer() :
 ^~~~~
[ 33%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/MonsterFleetPlansParser.cpp.o
[ 34%] Building CXX object CMakeFiles/freeorioncommon.dir/util/ScopedTimer.cpp.o
[ 34%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/Parse.cpp.o
[ 34%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SerializeEmpire.cpp.o
[ 35%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/PlanetEnvironmentValueRefParser.cpp.o
[ 35%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/PlanetSizeValueRefParser.cpp.o
[ 35%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SerializeModeratorAction.cpp.o
[ 36%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SerializeMultiplayerCommon.cpp.o
make[1]: *** [CMakeFiles/Makefile2:389: GG/CMakeFiles/GiGi.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 36%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SerializeOrderSet.cpp.o
In file included from /home/build/YPKG/root/freeorion/build/src-tarball/parse/ValueRefParserImpl.h:6:0,
                 from /home/build/YPKG/root/freeorion/build/src-tarball/parse/PlanetEnvironmentValueRefParser.cpp:1:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h: In member function ‘T ValueRef::Operation<T>::EvalImpl(const ScriptingContext&) const [with T = PlanetEnvironment]’:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h:1838:9: warning: statement will never be executed [-Wswitch-unreachable]
         if (m_operands.empty())
         ^~
In file included from /home/build/YPKG/root/freeorion/build/src-tarball/parse/ValueRefParserImpl.h:6:0,
                 from /home/build/YPKG/root/freeorion/build/src-tarball/parse/PlanetSizeValueRefParser.cpp:1:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h: In member function ‘T ValueRef::Operation<T>::EvalImpl(const ScriptingContext&) const [with T = PlanetSize]’:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h:1838:9: warning: statement will never be executed [-Wswitch-unreachable]
         if (m_operands.empty())
         ^~
[ 37%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SerializeUniverse.cpp.o
[ 37%] Building CXX object CMakeFiles/freeorioncommon.dir/util/SitRepEntry.cpp.o
[ 38%] Building CXX object CMakeFiles/freeorioncommon.dir/util/StringTable.cpp.o
[ 38%] Building CXX object CMakeFiles/freeorioncommon.dir/util/VarText.cpp.o
[ 38%] Building CXX object CMakeFiles/freeorioncommon.dir/util/Version.cpp.o
[ 39%] Building CXX object CMakeFiles/freeorioncommon.dir/util/XMLDoc.cpp.o
[ 39%] Building C object CMakeFiles/freeorioncommon.dir/util/binreloc.c.o
[ 40%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ReportParseError.cpp.o
[ 40%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/PlanetTypeValueRefParser.cpp.o
[ 40%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ShipDesignsParser.cpp.o
[ 40%] Linking CXX shared library libfreeorioncommon.so
[ 41%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ShipHullsParser.cpp.o
[ 41%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ShipPartsParser.cpp.o
In file included from /home/build/YPKG/root/freeorion/build/src-tarball/parse/ValueRefParserImpl.h:6:0,
                 from /home/build/YPKG/root/freeorion/build/src-tarball/parse/PlanetTypeValueRefParser.cpp:1:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h: In member function ‘T ValueRef::Operation<T>::EvalImpl(const ScriptingContext&) const [with T = PlanetType]’:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h:1838:9: warning: statement will never be executed [-Wswitch-unreachable]
         if (m_operands.empty())
         ^~
[ 41%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/SpecialsParser.cpp.o
[ 42%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/SpeciesParser.cpp.o
[ 42%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/StarTypeValueRefParser.cpp.o
[ 43%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/StringComplexValueRefParser.cpp.o
[ 43%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/StringValueRefParser.cpp.o
In file included from /home/build/YPKG/root/freeorion/build/src-tarball/parse/ValueRefParserImpl.h:6:0,
                 from /home/build/YPKG/root/freeorion/build/src-tarball/parse/StarTypeValueRefParser.cpp:1:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h: In member function ‘T ValueRef::Operation<T>::EvalImpl(const ScriptingContext&) const [with T = StarType]’:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h:1838:9: warning: statement will never be executed [-Wswitch-unreachable]
         if (m_operands.empty())
         ^~
[ 43%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/TechsParser.cpp.o
[ 44%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/Tokens.cpp.o
[ 44%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/UniverseObjectTypeValueRefParser.cpp.o
[ 45%] Building CXX object CMakeFiles/freeorionparseobj.dir/parse/ValueRefParserImpl.cpp.o
In file included from /home/build/YPKG/root/freeorion/build/src-tarball/parse/ValueRefParserImpl.h:6:0,
                 from /home/build/YPKG/root/freeorion/build/src-tarball/parse/UniverseObjectTypeValueRefParser.cpp:1:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h: In member function ‘T ValueRef::Operation<T>::EvalImpl(const ScriptingContext&) const [with T = UniverseObjectType]’:
/home/build/YPKG/root/freeorion/build/src-tarball/parse/../universe/ValueRef.h:1838:9: warning: statement will never be executed [-Wswitch-unreachable]
         if (m_operands.empty())
         ^~
[ 45%] Built target freeorionparseobj
[ 45%] Built target freeorioncommon
make: *** [Makefile:152: all] Error 2
```

### Expected Result
Build succeed as it used to previously (with boost 1.62.0)

### Steps to reproduce
* First step.
`cmake .`
* Second step.
`make`

**Comment by dbenage-cx** [comment]
> GG/src/dialogs/FileDlg.cpp:748:48: error: ‘boost::system::posix_error’ has not been declared

Addressed for 0.4.8 in c9b5b13f

Propose `won't fix` since boost v1.66.0 was not released before FreeOrion 0.4.7.1 (especially as 0.4.8 is approaching).

**Comment by kyrios123** [comment]
FreeOrion 0.4.7.1 successfully build with this patch. That's all I need ! 😄 
Thanks.
